File: src/main/java/com/dv/telegram/data/WikiBotCommandData.java
package com.dv.telegram.data;

import lombok.Data;
import org.apache.commons.lang3.StringUtils;

import java.util.List;

@Data
public class WikiBotCommandData {
    private String answer;
    private String wordsString;
    private List<String> words;

    public WikiBotCommandData(String answer, String wordsString, List<String> words) { // todo: Kotlin + Lombok hack
        this.answer = answer;
        this.wordsString = wordsString;
        this.words = words;
    }

    public boolean isPresentIn(String text) {
        if (StringUtils.isBlank(text)) {
            return false;
        }

        for (String word : words) {
            if (text.contains(word)) {
                return true;
            }
        }

        return false;
    }

    public String getOneLineAnswer() {
        return answer;
    }

    public String getMultiLineAnswer() {
        return String.format("— %s", answer);
    }
}

File: src/main/kotlin/com/dv/telegram/data/WikiBotCommandData.kt
package com.dv.telegram.data

data class WikiBotCommandData(
    val answer: String,
    val wordsString: String,
    val words: List<String>
) : BotAnswerData {
    override fun isPresentIn(text: String) = isPresentIn(text, words)

    override fun getMatch(text: String) = getMatch(text, words)

    fun getAnswerWithoutBullet() = answer

    fun getAnswerWithBullet(bullet: String) = "$bullet $answer"
}
